modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/system/info/item/provider/test/SystemObjectEntryInfoItemDetailsProviderTest.java

```diff
@@ -0,0 +1,198 @@
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = _groupLocalService.fetchGroup(TestPropsValues.getGroupId());
+
+		ServiceContextThreadLocal.pushServiceContext(_getServiceContext());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ServiceContextThreadLocal.popServiceContext();
+	}
+
+	@Test
+	public void testGetInfoItem() throws Exception {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				TestPropsValues.getCompanyId(), User.class.getSimpleName());
+
+		SystemObjectDefinitionManager systemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager(objectDefinition.getName());
+
+		long userId = systemObjectDefinitionManager.addBaseModel(
+			TestPropsValues.getUser(),
+			HashMapBuilder.<String, Object>put(
+				"alternateName", "john"
+			).put(
+				"emailAddress", "john@liferay.com"
+			).put(
+				"familyName", "Smith"
+			).put(
+				"givenName", "John"
+			).build());
+
+		String objectDefinitionItemClassName = StringBundler.concat(
+			objectDefinition.getClassName(), StringPool.POUND,
+			objectDefinition.getObjectDefinitionId());
+
+		InfoItemObjectProvider<SystemObjectEntry> infoItemObjectProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoItemObjectProvider.class, objectDefinitionItemClassName);
+
+		SystemObjectEntry systemObjectEntry =
+			infoItemObjectProvider.getInfoItem(
+				new ClassPKInfoItemIdentifier(userId));
+
+		InfoItemDetailsProvider<Object> infoItemDetailsProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoItemDetailsProvider.class, objectDefinitionItemClassName);
+
+		InfoItemDetails infoItemDetails =
+			infoItemDetailsProvider.getInfoItemDetails(systemObjectEntry);
+
+		Assert.assertEquals(
+			objectDefinitionItemClassName, infoItemDetails.getClassName());
+		Assert.assertEquals(
+			new InfoItemReference(
+				objectDefinitionItemClassName, systemObjectEntry.getClassPK()),
+			infoItemDetails.getInfoItemReference());
+
+		Map<String, Object> values = systemObjectEntry.getValues();
+
+		String externalReferenceCode = String.valueOf(
+			values.get("externalReferenceCode"));
+
+		systemObjectEntry = infoItemObjectProvider.getInfoItem(
+			_group.getGroupId(),
+			new ERCInfoItemIdentifier(
+				externalReferenceCode, _group.getExternalReferenceCode()));
+
+		infoItemDetails = infoItemDetailsProvider.getInfoItemDetails(
+			systemObjectEntry);
+
+		Assert.assertEquals(
+			objectDefinitionItemClassName, infoItemDetails.getClassName());
+		Assert.assertEquals(
+			new InfoItemReference(
+				objectDefinitionItemClassName, systemObjectEntry.getClassPK()),
+			infoItemDetails.getInfoItemReference());
```

Comment:
I don’t think this scenario is valuable because `getInfoItemDetails(SystemObjectEntry systemObjectEntry) `returns based on `ClassPKInfoItemIdentifier` (which you already tested earlier).